### PR TITLE
acquire/release token on TaskWorker without errors

### PR DESCRIPTION
**Pull request by jakekaplan**
When acquiring or releasing a token we need to be sure to not propagate `RuntimeError`s so that the `TaskWorker` keeps running.

### CODE REVIEW

File: src/prefect/task_worker.py [hunk 1]
```diff
@@ -157,11 +178,16 @@ async def _subscribe_to_task_scheduling(self):
...
-                self._safe_submit_scheduled_task_run, task_run
-            )
+
+            token_acquired = await self._acquire_token(task_run.id)
+            if token_acquired:
+                self._runs_task_group.start_soon(
+                    self._safe_submit_scheduled_task_run, task_run
+                )
+            else:
+                logger.info(
```
Comment (jakekaplan) [inline src/prefect/task_worker.py:188]: Maybe this shouldn't be an info level log

**Review by desertaxle — COMMENTED** [review]
Can you add a test for this? Submitting the same task run twice via `execute_task_run` should give us the confidence we need.

**Comment by jakekaplan** [comment]
> Can you add a test for this? Submitting the same task run twice via `execute_task_run` should give us the confidence we need.

Added!

File: tests/test_task_worker.py [hunk 2]
```diff
@@ -672,6 +672,36 @@ async def register_localfilesystem(self):
...
+        task_run = await prefect_client.read_task_run(task_run_future.task_run_id)
+
+        try:
+            with anyio.move_on_after(1):
+                # run same task, one should acquire a token
+                # the other will gracefully be skipped.
+                async with task_worker:
+                    await asyncio.gather(
+                        task_worker.execute_task_run(task_run),
+                        task_worker.execute_task_run(task_run),
```
Comment (jakekaplan) [inline tests/test_task_worker.py:696]: On `main` this errors with `this borrower is already holding one of this CapacityLimiter's tokens`

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**